### PR TITLE
Skip freeze_time tests in server mode

### PR DESCRIPTION
**Pull request by garyd203**
Skip tests that use `freeze_time` when testing server mode. This is because freeze_time is unable to affect the time on the external server process, which makes the tests fragile and liable to incorrectly fail on a time comparison

### CODE REVIEW

File: tests/test_sqs/test_sqs.py [hunk 1]
```diff
@@ -837,9 +837,10 @@ def test_delete_message_after_visibility_timeout():
     assert new_queue.count() == 0
 
 
+@skip_in_server_mode
 @mock_sqs
 def test_batch_change_message_visibility():
-    if os.environ.get('TEST_SERVER_MODE', 'false').lower() == 'true':
+    if settings.TEST_SERVER_MODE:
```
Comment (terrycain) [inline tests/test_sqs/test_sqs.py]: If the function has been decorated with skip_test then you dont need the if statement right?

File: tests/test_sqs/test_sqs.py [hunk 2]
```diff
@@ -967,9 +968,10 @@ def test_create_fifo_queue_with_dlq():
         )
 
 
+@skip_in_server_mode
 @mock_sqs
 def test_queue_with_dlq():
-    if os.environ.get('TEST_SERVER_MODE', 'false').lower() == 'true':
+    if settings.TEST_SERVER_MODE:
```
Comment (terrycain) [inline tests/test_sqs/test_sqs.py]: As above

File: tests/test_sqs/test_sqs.py [hunk 3]
```diff
@@ -1153,9 +1155,10 @@ def test_receive_messages_with_message_group_id_on_requeue():
     messages[0].message_id.should.equal(message.message_id)
 
 
+@skip_in_server_mode
 @mock_sqs
 def test_receive_messages_with_message_group_id_on_visibility_timeout():
-    if os.environ.get('TEST_SERVER_MODE', 'false').lower() == 'true':
+    if settings.TEST_SERVER_MODE:
```
Comment (terrycain) [inline tests/test_sqs/test_sqs.py]: As above

**Comment by terrycain** [comment]
Other than those changes, this looks good. @JackDanger 2nd opinion

**Comment by garyd203** [comment]
Thanks @terrycain, I have stripped out those duplicate checks now

**Comment by garyd203** [comment]
hi @terrycain, is there anything else we need to do to get this change accepted? Ta!

**Comment by garyd203** [comment]
Hi @terrycain Are we able to merge this PR now?

**Comment by garyd203** [comment]
Hi @spulec and @terrycain Is there anything else I can do to get this patch accepted? If it is no longer relevant I am happy to abandon it.

Thanks heaps (and thanks for continuing to maintain moto).

**Comment by spulec** [comment]
I'm a little worried about this. There are legitimate differences between server-mode and non-server mode and I'm worried about the loss of coverage here.

Is there another approach? Setting the time to freeze as an environment variable or something?

**Comment by bpandola** [comment]
There have been a lot of updates to the codebase since this was opened, including a move from `nosetests` (upon which this new `@skip_in_server_mode` decorator had been based) to `pytest`.  Most of the tests that use `@freeze_time` have had explicit `TEST_SERVER_MODE` checks added over the years, which obviates the need to try to bring this PR up-to-date and fix the long list of now-conflicting files.

**Comment by garyd203** [comment]
Thanks @bpandola 👍

**Pull request closed without merging**